package/MDAnalysis/analysis/density.py

```diff
@@ -229,6 +229,13 @@ class DensityAnalysis(AnalysisBase):
     are within 4 Å of the protein heavy atoms) then create an
     :class:`~MDAnalysis.core.groups.UpdatingAtomGroup` (see Examples).
 
+    If the 'AtomGroup' instance does not contain any selection of atoms and 
+    'updating' is set to true, an empty :class: UpdatingAtomGroup is returned.
+    If an instance of this class is passed onto DensityAnalysis without any 
+    user-defined grid, DensityAnalysis will fail because it cannot predict 
+    selection bounds for future frames in the trajectory. In such a situation, 
+    user defined box limits should be provided ensuring that the provided 
+    limits encompass all atoms in selection on future frames.
```

Comment:
This text does not reflect what the code does, or at least that's not clear. The only change that you made is to emit warnings, right? 

State what the problem can be (empty `atomgroup` – note write this in reST as
```reST
When the `atomgroup` is empty when :class:`DensityAnalysis` is initialized ...
```
so that you're referring to variables in the doc string). Then state what happens (warning emitted). Finally explain what the user should be doing (i.e., repeat what the warning says but elaborate). Imagine that you're the user of the function: What do you need to know when you

1. want to run DensityAnalysis for the first time
2. when you run into an error and think "maybe the docs tell me what to do because this error message was very confusing"

Rewrite with the above in mind.

The sentence 
> "If the 'AtomGroup' instance does not contain any selection of atoms and  'updating' is set to true, an empty :class: UpdatingAtomGroup is returned."

is confusing because the problem will occur as soon as `atomgroup` is empty, regardless of being updating or not. It's just that the situation is more likely to happen with an `UpdatingAtomGroup`.


Maybe @TBGAnsell has some suggestions what they _would_ have liked to see (see #3055)? Comments welcome!


For exceptions that are raised and that need explanation, add a section (after Returns)

```reST
Raises
------
ValueError: raised when ....
```

Reply:
I've given it some thought and updated the Docstring. I tried to keep it as simple as possible so that a user will know what to do! :)

Reply:
Hello, the message here seems clear to me. One further suggestion could be to output the frames in which the density analysis fails, similar to the frames reported when H-bonds are not detected in PropkaTraj when skip_failure=True. This could be useful for an initial quick scan to see eg: how much equilibration time is needed for the ligand to reach the site.

Reply:
Okay, maybe I'll include the frame number in the warning message!

Reply:
The warning will only get called once at construction, so it won't do what @TBGAnsell is looking for.

That being said, whilst I'm not against the idea of including the frame number in the warning / error, I am somewhat opposed to checking every frame for the presence of atoms in the selection. My view is twofold:
1) An empty UpdatingAtomGroup at a given frame isn't necessarily a failure (unlike for PropkaTraj where we know that it's a failure [note: skip_failure captures a non-specific propka 3.x failure, not just solely the failure to detect any specific thing]).
2) Since we aren't currently checking for it, it will add extra checking code that probablyl won't be cheap. In my opinion, if folks want to do a quick scan it's much cheaper/cleaner to just do a quick loop over your updatingatomgroup.

That being said, I'm willing to default to @orbeckst's opinion on the matter.

Reply:
Thank you @TBGAnsell for your comments — much appreciated!

I agree with @IAlibay reservations about checking every frame because of likely performance reasons. If a "scanning" feature is desired then this can be a new issue (please raise it and we discuss there). Let's keep this one focused on the original problem.